mvapich2@2.3.7 fabrics=mrail: tools/bootstrap/src/bsci_init.c: 'HYDT_bsci_launcher_pbs_init' undeclared here (not in a function)

### Steps to reproduce the issue

Installing `mvapich2@2.3.7 fabrics=mrail ~wrapperrpath` is failing here using:
* `spack@develop` (b260234faf34b19227e87698bc2a1402c4ea76ac from `Tue Apr 18 03:42:56 2023 -0700`)
* GCC 11.1.0
* Ubuntu 20.04, X86_64

Spack environment:
```
spack:
  view: false

  concretizer:
    reuse: false
    unify: false

  packages:
    all:
      compiler: [gcc@11.1.0]
      providers:
        blas: [openblas]
        mpi: [mvapich2]
      target: [x86_64]
    xz:
      require: "+pic"
    ncurses:
      require: '@6.3 +termlib'
    rdma-core:
      buildable: false
      externals:
      - spec: rdma-core@44.0
        prefix: /usr

  compilers:
  - compiler:
      spec: gcc@11.1.0
      paths:
        cc: /usr/bin/gcc
        cxx: /usr/bin/g++
        f77: /usr/bin/gfortran
        fc: /usr/bin/gfortran
      flags: {}
      operating_system: ubuntu20.04
      target: x86_64
      modules: []
      environment: {}
      extra_rpaths: []

  specs:
  - mvapich2 fabrics=mrail ~wrapperrpath
```

<details><summary>Concretization</summary>
<pre>
 -   w7eusht  mvapich2@2.3.7%gcc@11.1.0~alloca~cuda~debug~hwlocv2+regcache~wrapperrpath build_system=autotools ch3_rank_bits=32 fabrics=mrail file_systems=auto process_managers=auto threads=multiple arch=linux-ubuntu20.04-x86_64
[+]  pkp6o3z      ^bison@3.8.2%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  j2tlgt4          ^diffutils@3.9%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  e6fw62o          ^m4@1.4.19%gcc@11.1.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-ubuntu20.04-x86_64
[+]  b2tzsvo              ^libsigsegv@2.14%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  roah3c6          ^perl@5.36.0%gcc@11.1.0+cpanm+open+shared+threads build_system=generic arch=linux-ubuntu20.04-x86_64
[+]  alkzfee              ^berkeley-db@18.1.40%gcc@11.1.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-ubuntu20.04-x86_64
[+]  onsas2b              ^bzip2@1.0.8%gcc@11.1.0~debug~pic+shared build_system=generic arch=linux-ubuntu20.04-x86_64
[+]  qkfi67z              ^gdbm@1.23%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  lui3zbo                  ^readline@8.2%gcc@11.1.0 build_system=autotools patches=bbf97f1 arch=linux-ubuntu20.04-x86_64
[+]  lwdu2k7                      ^ncurses@6.3%gcc@11.1.0~symlinks+termlib abi=none build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  bqre6so      ^findutils@4.9.0%gcc@11.1.0 build_system=autotools patches=440b954 arch=linux-ubuntu20.04-x86_64
[+]  3mmidga      ^libpciaccess@0.17%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  kagjqhf          ^libtool@2.4.7%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  i5p6x5b          ^util-macros@1.19.3%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  nsmovt5      ^libxml2@2.10.3%gcc@11.1.0~python build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  5f66p4a          ^libiconv@1.17%gcc@11.1.0 build_system=autotools libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]  tppbn4x          ^xz@5.4.1%gcc@11.1.0+pic build_system=autotools libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]  uorrjkf      ^pkgconf@1.8.0%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  mr2eijk      ^rdma-core@44.0%gcc@11.1.0+static build_system=cmake build_type=RelWithDebInfo generator=make arch=linux-ubuntu20.04-x86_64
[+]  hi2xsgc      ^zlib@1.2.13%gcc@11.1.0+optimize+pic+shared build_system=makefile arch=linux-ubuntu20.04-x86_64
</pre></details>

Error:
```
$> spack -e . install
...
==> Installing mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm
==> No binary for mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm found: installing from source
==> Fetching https://mvapich.cse.ohio-state.edu/download/mvapich/mv2/mvapich2-2.3.7.tar.gz
==> No patches needed for mvapich2
==> mvapich2: Executing phase: 'autoreconf'
==> mvapich2: Executing phase: 'configure'
==> mvapich2: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' 'V=1'

5 errors found in build log:
     4181    libtool: compile:  /opt/mvapich2/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I./include -I./tools/topo/hwloc/hwloc/include/private/autogen -I./tools/topo/hwloc/hwloc/include
             /hwloc/autogen -I./include -I./mpl/include -I./mpl/include -I./tools/topo -I./tools/topo/hwloc/hwloc/include -I./tools/topo/hwloc/hwloc/include -I./tools/bootstrap/include -I./to
             ols/bootstrap/utils -I./tools/ckpoint -I./tools/demux -I./tools/debugger -I./pm/include -I./pm/utils -I./ui/include -I./ui/utils -DHYDRA_CONF_FILE=\"/opt/mvapich2/spack/opt/spack
             /linux-ubuntu20.04-x86_64/gcc-11.1.0/mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/etc/mpiexec.hydra.conf\" -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkop
             cjfesmuz572yehm/spack-src/src/pm/hydra/tools/topo/hwloc/hwloc/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/pm/hydra/t
             ools/topo/hwloc/hwloc/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpl/include -I/tmp/root/spack-stage/spack-stage-mv
             apich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpl/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/openpa/s
             rc -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/openpa/src -D_REENTRANT -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7
             eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpi/romio/include -fvisibility=hidden -I/opt/mvapich2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libpciaccess-0.17-3mmidgaul
             eyk7arfikq33raxymzuraus/include -I/opt/mvapich2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libxml2-2.10.3-nsmovt5lnxxavuiftxfxdobd4do4ynup/include/libxml2 -I/opt/mvapich
             2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libiconv-1.17-5f66p4aoivmlxa7htmpsffk5ov62vgci/include -DNDEBUG -DNVALGRIND -O2 -g -MT tools/topo/topo.lo -MD -MP -MF tools/
             topo/.deps/topo.Tpo -c tools/topo/topo.c  -fPIC -DPIC -o tools/topo/.libs/topo.o
     4182    libtool: compile:  /opt/mvapich2/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I./include -I./tools/topo/hwloc/hwloc/include/private/autogen -I./tools/topo/hwloc/hwloc/include
             /hwloc/autogen -I./include -I./mpl/include -I./mpl/include -I./tools/topo -I./tools/topo/hwloc/hwloc/include -I./tools/topo/hwloc/hwloc/include -I./tools/bootstrap/include -I./to
             ols/bootstrap/utils -I./tools/ckpoint -I./tools/demux -I./tools/debugger -I./pm/include -I./pm/utils -I./ui/include -I./ui/utils -DHYDRA_CONF_FILE=\"/opt/mvapich2/spack/opt/spack
             /linux-ubuntu20.04-x86_64/gcc-11.1.0/mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/etc/mpiexec.hydra.conf\" -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkop
             cjfesmuz572yehm/spack-src/src/pm/hydra/tools/topo/hwloc/hwloc/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/pm/hydra/t
             ools/topo/hwloc/hwloc/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpl/include -I/tmp/root/spack-stage/spack-stage-mv
             apich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpl/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/openpa/s
             rc -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/openpa/src -D_REENTRANT -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7
             eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpi/romio/include -fvisibility=hidden -I/opt/mvapich2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libpciaccess-0.17-3mmidgaul
             eyk7arfikq33raxymzuraus/include -I/opt/mvapich2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libxml2-2.10.3-nsmovt5lnxxavuiftxfxdobd4do4ynup/include/libxml2 -I/opt/mvapich
             2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libiconv-1.17-5f66p4aoivmlxa7htmpsffk5ov62vgci/include -DNDEBUG -DNVALGRIND -O2 -g -MT tools/bootstrap/utils/bscu_wait.lo -M
             D -MP -MF tools/bootstrap/utils/.deps/bscu_wait.Tpo -c tools/bootstrap/utils/bscu_wait.c  -fPIC -DPIC -o tools/bootstrap/utils/.libs/bscu_wait.o
     4183    libtool: compile:  /opt/mvapich2/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I./include -I./tools/topo/hwloc/hwloc/include/private/autogen -I./tools/topo/hwloc/hwloc/include
             /hwloc/autogen -I./include -I./mpl/include -I./mpl/include -I./tools/topo -I./tools/topo/hwloc/hwloc/include -I./tools/topo/hwloc/hwloc/include -I./tools/bootstrap/include -I./to
             ols/bootstrap/utils -I./tools/ckpoint -I./tools/demux -I./tools/debugger -I./pm/include -I./pm/utils -I./ui/include -I./ui/utils -DHYDRA_CONF_FILE=\"/opt/mvapich2/spack/opt/spack
             /linux-ubuntu20.04-x86_64/gcc-11.1.0/mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/etc/mpiexec.hydra.conf\" -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkop
             cjfesmuz572yehm/spack-src/src/pm/hydra/tools/topo/hwloc/hwloc/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/pm/hydra/t
             ools/topo/hwloc/hwloc/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpl/include -I/tmp/root/spack-stage/spack-stage-mv
             apich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpl/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/openpa/s
             rc -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/openpa/src -D_REENTRANT -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7
             eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpi/romio/include -fvisibility=hidden -I/opt/mvapich2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libpciaccess-0.17-3mmidgaul
             eyk7arfikq33raxymzuraus/include -I/opt/mvapich2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libxml2-2.10.3-nsmovt5lnxxavuiftxfxdobd4do4ynup/include/libxml2 -I/opt/mvapich
             2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libiconv-1.17-5f66p4aoivmlxa7htmpsffk5ov62vgci/include -DNDEBUG -DNVALGRIND -O2 -g -MT utils/signals/signals.lo -MD -MP -MF
             utils/signals/.deps/signals.Tpo -c utils/signals/signals.c  -fPIC -DPIC -o utils/signals/.libs/signals.o
     4184    libtool: compile:  /opt/mvapich2/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I./include -I./tools/topo/hwloc/hwloc/include/private/autogen -I./tools/topo/hwloc/hwloc/include
             /hwloc/autogen -I./include -I./mpl/include -I./mpl/include -I./tools/topo -I./tools/topo/hwloc/hwloc/include -I./tools/topo/hwloc/hwloc/include -I./tools/bootstrap/include -I./to
             ols/bootstrap/utils -I./tools/ckpoint -I./tools/demux -I./tools/debugger -I./pm/include -I./pm/utils -I./ui/include -I./ui/utils -DHYDRA_CONF_FILE=\"/opt/mvapich2/spack/opt/spack
             /linux-ubuntu20.04-x86_64/gcc-11.1.0/mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/etc/mpiexec.hydra.conf\" -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkop
             cjfesmuz572yehm/spack-src/src/pm/hydra/tools/topo/hwloc/hwloc/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/pm/hydra/t
             ools/topo/hwloc/hwloc/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpl/include -I/tmp/root/spack-stage/spack-stage-mv
             apich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpl/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/openpa/s
             rc -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/openpa/src -D_REENTRANT -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7
             eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpi/romio/include -fvisibility=hidden -I/opt/mvapich2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libpciaccess-0.17-3mmidgaul
             eyk7arfikq33raxymzuraus/include -I/opt/mvapich2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libxml2-2.10.3-nsmovt5lnxxavuiftxfxdobd4do4ynup/include/libxml2 -I/opt/mvapich
             2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libiconv-1.17-5f66p4aoivmlxa7htmpsffk5ov62vgci/include -DNDEBUG -DNVALGRIND -O2 -g -MT tools/bootstrap/src/bsci_query_proxy_
             id.lo -MD -MP -MF tools/bootstrap/src/.deps/bsci_query_proxy_id.Tpo -c tools/bootstrap/src/bsci_query_proxy_id.c  -fPIC -DPIC -o tools/bootstrap/src/.libs/bsci_query_proxy_id.o
     4185    libtool: compile:  /opt/mvapich2/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I./include -I./tools/topo/hwloc/hwloc/include/private/autogen -I./tools/topo/hwloc/hwloc/include
             /hwloc/autogen -I./include -I./mpl/include -I./mpl/include -I./tools/topo -I./tools/topo/hwloc/hwloc/include -I./tools/topo/hwloc/hwloc/include -I./tools/bootstrap/include -I./to
             ols/bootstrap/utils -I./tools/ckpoint -I./tools/demux -I./tools/debugger -I./pm/include -I./pm/utils -I./ui/include -I./ui/utils -DHYDRA_CONF_FILE=\"/opt/mvapich2/spack/opt/spack
             /linux-ubuntu20.04-x86_64/gcc-11.1.0/mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/etc/mpiexec.hydra.conf\" -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkop
             cjfesmuz572yehm/spack-src/src/pm/hydra/tools/topo/hwloc/hwloc/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/pm/hydra/t
             ools/topo/hwloc/hwloc/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpl/include -I/tmp/root/spack-stage/spack-stage-mv
             apich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpl/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/openpa/s
             rc -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/openpa/src -D_REENTRANT -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7
             eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpi/romio/include -fvisibility=hidden -I/opt/mvapich2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libpciaccess-0.17-3mmidgaul
             eyk7arfikq33raxymzuraus/include -I/opt/mvapich2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libxml2-2.10.3-nsmovt5lnxxavuiftxfxdobd4do4ynup/include/libxml2 -I/opt/mvapich
             2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libiconv-1.17-5f66p4aoivmlxa7htmpsffk5ov62vgci/include -DNDEBUG -DNVALGRIND -O2 -g -MT tools/bootstrap/external/fork_init.lo
              -MD -MP -MF tools/bootstrap/external/.deps/fork_init.Tpo -c tools/bootstrap/external/fork_init.c  -fPIC -DPIC -o tools/bootstrap/external/.libs/fork_init.o
     4186    libtool: compile:  /opt/mvapich2/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I./include -I./tools/topo/hwloc/hwloc/include/private/autogen -I./tools/topo/hwloc/hwloc/include
             /hwloc/autogen -I./include -I./mpl/include -I./mpl/include -I./tools/topo -I./tools/topo/hwloc/hwloc/include -I./tools/topo/hwloc/hwloc/include -I./tools/bootstrap/include -I./to
             ols/bootstrap/utils -I./tools/ckpoint -I./tools/demux -I./tools/debugger -I./pm/include -I./pm/utils -I./ui/include -I./ui/utils -DHYDRA_CONF_FILE=\"/opt/mvapich2/spack/opt/spack
             /linux-ubuntu20.04-x86_64/gcc-11.1.0/mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/etc/mpiexec.hydra.conf\" -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkop
             cjfesmuz572yehm/spack-src/src/pm/hydra/tools/topo/hwloc/hwloc/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/pm/hydra/t
             ools/topo/hwloc/hwloc/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpl/include -I/tmp/root/spack-stage/spack-stage-mv
             apich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpl/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/openpa/s
             rc -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/openpa/src -D_REENTRANT -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7
             eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpi/romio/include -fvisibility=hidden -I/opt/mvapich2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libpciaccess-0.17-3mmidgaul
             eyk7arfikq33raxymzuraus/include -I/opt/mvapich2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libxml2-2.10.3-nsmovt5lnxxavuiftxfxdobd4do4ynup/include/libxml2 -I/opt/mvapich
             2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libiconv-1.17-5f66p4aoivmlxa7htmpsffk5ov62vgci/include -DNDEBUG -DNVALGRIND -O2 -g -MT utils/string/string.lo -MD -MP -MF ut
             ils/string/.deps/string.Tpo -c utils/string/string.c  -fPIC -DPIC -o utils/string/.libs/string.o
  >> 4187    tools/bootstrap/src/bsci_init.c:17:266: error: 'HYDT_bsci_launcher_pbs_init' undeclared here (not in a function); did you mean 'HYDT_bsci_launcher_sge_init'?
     4188       17 | static HYD_status(*launcher_init_array[])(void) = {  HYDT_bsci_launcher_ssh_init,  HYDT_bsci_launcher_rsh_init,  HYDT_bsci_launcher_fork_init,  HYDT_bsci_launcher_slurm_i
             nit,  HYDT_bsci_launcher_ll_init,  HYDT_bsci_launcher_lsf_init,  HYDT_bsci_launcher_sge_init,  HYDT_bsci_launcher_pbs_init,  HYDT_bsci_launcher_manual_init,  HYDT_bsci_launcher_p
             ersist_init,  NULL };
     4189          |
                                                                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~
     4190          |
                                                                                                            HYDT_bsci_launcher_sge_init
  >> 4191    make[4]: *** [Makefile:1595: tools/bootstrap/src/bsci_init.lo] Error 1
     4192    make[4]: *** Waiting for unfinished jobs....
     4193    In file included from /usr/include/stdio.h:867,
     4194                     from ./mpl/include/mpl.h:12,
     4195                     from ./include/hydra.h:17,
     4196                     from utils/dbg/dbg.c:7:
     4197    In function 'fprintf',

     ...

     4229    libtool: compile:  /opt/mvapich2/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I./include -I./tools/topo/hwloc/hwloc/include/private/autogen -I./tools/topo/hwloc/hwloc/include
             /hwloc/autogen -I./include -I./mpl/include -I./mpl/include -I./tools/topo -I./tools/topo/hwloc/hwloc/include -I./tools/topo/hwloc/hwloc/include -I./tools/bootstrap/include -I./to
             ols/bootstrap/utils -I./tools/ckpoint -I./tools/demux -I./tools/debugger -I./pm/include -I./pm/utils -I./ui/include -I./ui/utils -DHYDRA_CONF_FILE=\"/opt/mvapich2/spack/opt/spack
             /linux-ubuntu20.04-x86_64/gcc-11.1.0/mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/etc/mpiexec.hydra.conf\" -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkop
             cjfesmuz572yehm/spack-src/src/pm/hydra/tools/topo/hwloc/hwloc/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/pm/hydra/t
             ools/topo/hwloc/hwloc/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpl/include -I/tmp/root/spack-stage/spack-stage-mv
             apich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpl/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/openpa/s
             rc -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/openpa/src -D_REENTRANT -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7
             eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpi/romio/include -fvisibility=hidden -I/opt/mvapich2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libpciaccess-0.17-3mmidgaul
             eyk7arfikq33raxymzuraus/include -I/opt/mvapich2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libxml2-2.10.3-nsmovt5lnxxavuiftxfxdobd4do4ynup/include/libxml2 -I/opt/mvapich
             2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libiconv-1.17-5f66p4aoivmlxa7htmpsffk5ov62vgci/include -DNDEBUG -DNVALGRIND -O2 -g -MT utils/string/string.lo -MD -MP -MF ut
             ils/string/.deps/string.Tpo -c utils/string/string.c -o utils/string/string.o >/dev/null 2>&1
     4230    libtool: compile:  /opt/mvapich2/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I./include -I./tools/topo/hwloc/hwloc/include/private/autogen -I./tools/topo/hwloc/hwloc/include
             /hwloc/autogen -I./include -I./mpl/include -I./mpl/include -I./tools/topo -I./tools/topo/hwloc/hwloc/include -I./tools/topo/hwloc/hwloc/include -I./tools/bootstrap/include -I./to
             ols/bootstrap/utils -I./tools/ckpoint -I./tools/demux -I./tools/debugger -I./pm/include -I./pm/utils -I./ui/include -I./ui/utils -DHYDRA_CONF_FILE=\"/opt/mvapich2/spack/opt/spack
             /linux-ubuntu20.04-x86_64/gcc-11.1.0/mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/etc/mpiexec.hydra.conf\" -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkop
             cjfesmuz572yehm/spack-src/src/pm/hydra/tools/topo/hwloc/hwloc/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/pm/hydra/t
             ools/topo/hwloc/hwloc/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpl/include -I/tmp/root/spack-stage/spack-stage-mv
             apich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpl/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/openpa/s
             rc -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/openpa/src -D_REENTRANT -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7
             eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpi/romio/include -fvisibility=hidden -I/opt/mvapich2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libpciaccess-0.17-3mmidgaul
             eyk7arfikq33raxymzuraus/include -I/opt/mvapich2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libxml2-2.10.3-nsmovt5lnxxavuiftxfxdobd4do4ynup/include/libxml2 -I/opt/mvapich
             2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libiconv-1.17-5f66p4aoivmlxa7htmpsffk5ov62vgci/include -DNDEBUG -DNVALGRIND -O2 -g -MT utils/args/args.lo -MD -MP -MF utils/
             args/.deps/args.Tpo -c utils/args/args.c -o utils/args/args.o >/dev/null 2>&1
     4231    libtool: compile:  /opt/mvapich2/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I./include -I./tools/topo/hwloc/hwloc/include/private/autogen -I./tools/topo/hwloc/hwloc/include
             /hwloc/autogen -I./include -I./mpl/include -I./mpl/include -I./tools/topo -I./tools/topo/hwloc/hwloc/include -I./tools/topo/hwloc/hwloc/include -I./tools/bootstrap/include -I./to
             ols/bootstrap/utils -I./tools/ckpoint -I./tools/demux -I./tools/debugger -I./pm/include -I./pm/utils -I./ui/include -I./ui/utils -DHYDRA_CONF_FILE=\"/opt/mvapich2/spack/opt/spack
             /linux-ubuntu20.04-x86_64/gcc-11.1.0/mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/etc/mpiexec.hydra.conf\" -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkop
             cjfesmuz572yehm/spack-src/src/pm/hydra/tools/topo/hwloc/hwloc/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/pm/hydra/t
             ools/topo/hwloc/hwloc/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpl/include -I/tmp/root/spack-stage/spack-stage-mv
             apich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpl/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/openpa/s
             rc -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/openpa/src -D_REENTRANT -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7
             eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpi/romio/include -fvisibility=hidden -I/opt/mvapich2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libpciaccess-0.17-3mmidgaul
             eyk7arfikq33raxymzuraus/include -I/opt/mvapich2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libxml2-2.10.3-nsmovt5lnxxavuiftxfxdobd4do4ynup/include/libxml2 -I/opt/mvapich
             2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libiconv-1.17-5f66p4aoivmlxa7htmpsffk5ov62vgci/include -DNDEBUG -DNVALGRIND -O2 -g -MT tools/bootstrap/external/external_com
             mon_launch.lo -MD -MP -MF tools/bootstrap/external/.deps/external_common_launch.Tpo -c tools/bootstrap/external/external_common_launch.c -o tools/bootstrap/external/external_comm
             on_launch.o >/dev/null 2>&1
     4232    libtool: compile:  /opt/mvapich2/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I./include -I./tools/topo/hwloc/hwloc/include/private/autogen -I./tools/topo/hwloc/hwloc/include
             /hwloc/autogen -I./include -I./mpl/include -I./mpl/include -I./tools/topo -I./tools/topo/hwloc/hwloc/include -I./tools/topo/hwloc/hwloc/include -I./tools/bootstrap/include -I./to
             ols/bootstrap/utils -I./tools/ckpoint -I./tools/demux -I./tools/debugger -I./pm/include -I./pm/utils -I./ui/include -I./ui/utils -DHYDRA_CONF_FILE=\"/opt/mvapich2/spack/opt/spack
             /linux-ubuntu20.04-x86_64/gcc-11.1.0/mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/etc/mpiexec.hydra.conf\" -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkop
             cjfesmuz572yehm/spack-src/src/pm/hydra/tools/topo/hwloc/hwloc/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/pm/hydra/t
             ools/topo/hwloc/hwloc/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpl/include -I/tmp/root/spack-stage/spack-stage-mv
             apich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpl/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/openpa/s
             rc -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/openpa/src -D_REENTRANT -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7
             eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpi/romio/include -fvisibility=hidden -I/opt/mvapich2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libpciaccess-0.17-3mmidgaul
             eyk7arfikq33raxymzuraus/include -I/opt/mvapich2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libxml2-2.10.3-nsmovt5lnxxavuiftxfxdobd4do4ynup/include/libxml2 -I/opt/mvapich
             2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libiconv-1.17-5f66p4aoivmlxa7htmpsffk5ov62vgci/include -DNDEBUG -DNVALGRIND -O2 -g -MT utils/sock/sock.lo -MD -MP -MF utils/
             sock/.deps/sock.Tpo -c utils/sock/sock.c -o utils/sock/sock.o >/dev/null 2>&1
     4233    libtool: compile:  /opt/mvapich2/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I./include -I./tools/topo/hwloc/hwloc/include/private/autogen -I./tools/topo/hwloc/hwloc/include
             /hwloc/autogen -I./include -I./mpl/include -I./mpl/include -I./tools/topo -I./tools/topo/hwloc/hwloc/include -I./tools/topo/hwloc/hwloc/include -I./tools/bootstrap/include -I./to
             ols/bootstrap/utils -I./tools/ckpoint -I./tools/demux -I./tools/debugger -I./pm/include -I./pm/utils -I./ui/include -I./ui/utils -DHYDRA_CONF_FILE=\"/opt/mvapich2/spack/opt/spack
             /linux-ubuntu20.04-x86_64/gcc-11.1.0/mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/etc/mpiexec.hydra.conf\" -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkop
             cjfesmuz572yehm/spack-src/src/pm/hydra/tools/topo/hwloc/hwloc/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/pm/hydra/t
             ools/topo/hwloc/hwloc/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpl/include -I/tmp/root/spack-stage/spack-stage-mv
             apich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpl/include -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/openpa/s
             rc -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/openpa/src -D_REENTRANT -I/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7
             eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/mpi/romio/include -fvisibility=hidden -I/opt/mvapich2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libpciaccess-0.17-3mmidgaul
             eyk7arfikq33raxymzuraus/include -I/opt/mvapich2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libxml2-2.10.3-nsmovt5lnxxavuiftxfxdobd4do4ynup/include/libxml2 -I/opt/mvapich
             2/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/libiconv-1.17-5f66p4aoivmlxa7htmpsffk5ov62vgci/include -DNDEBUG -DNVALGRIND -O2 -g -MT tools/topo/hwloc/topo_hwloc.lo -MD -M
             P -MF tools/topo/hwloc/.deps/topo_hwloc.Tpo -c tools/topo/hwloc/topo_hwloc.c -o tools/topo/hwloc/topo_hwloc.o >/dev/null 2>&1
     4234    make[4]: Leaving directory '/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/pm/hydra'
  >> 4235    make[3]: *** [Makefile:1765: all-recursive] Error 1
     4236    make[3]: Leaving directory '/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src/src/pm/hydra'
  >> 4237    make[2]: *** [Makefile:44393: all-recursive] Error 1
     4238    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src'
  >> 4239    make[1]: *** [Makefile:12029: all] Error 2
     4240    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-src'

See build log for details:
  /tmp/root/spack-stage/spack-stage-mvapich2-2.3.7-w7eushtwfsa23gkopcjfesmuz572yehm/spack-build-out.txt
```

### Error message

see above

### Information on your system

* **Spack:** 0.20.0.dev0 (7d37060434ea427f3e10a2fc91b7e35a38fb1577)
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-icelake
* **Concretizer:** clingo

### Additional information

[spack-build-env.txt](https://github.com/spack/spack/files/11267345/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/11267346/spack-build-out.txt)
[config.log](https://github.com/spack/spack/files/11267347/config.log)

@harisubramoni @natshineman @ndcontini

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/mvapich2/fix-torque.patch
+++ b/var/spack/repos/builtin/packages/mvapich2/fix-torque.patch
@@ -1,0 +1,27 @@
+diff -ruN spack-src/src/pm/hydra/configure.ac spack-src-patched/src/pm/hydra/configure.ac
+--- spack-src/src/pm/hydra/configure.ac	2022-05-16 16:58:22.000000000 +0000
++++ spack-src-patched/src/pm/hydra/configure.ac	2023-07-20 15:23:21.802299913 +0000
+@@ -306,15 +306,16 @@
+ 		    available_launchers="$available_launchers pbs"
+ 		    PAC_APPEND_FLAG([-ltorque],[WRAPPER_LIBS])
+                 fi
+-		PAC_SET_HEADER_LIB_PATH(pbs)
+ 		PAC_PUSH_FLAG(LIBS)
+-		PAC_CHECK_HEADER_LIB(tm.h, pbs, tm_init, have_pbs_launcher=yes,
++                if test "have_pbs_launcher" = "no" ; then
++		  PAC_CHECK_HEADER_LIB(tm.h, pbs, tm_init, have_pbs_launcher=yes,
+ 					   have_pbs_launcher=no)
+-		PAC_POP_FLAG(LIBS)
+-                if test "$have_pbs_launcher" = "yes" ; then
+-		    available_launchers="$available_launchers pbs"
+-		    PAC_APPEND_FLAG([-lpbs],[WRAPPER_LIBS])
+-                    AC_DEFINE(HAVE_PBS_PRO, 1, [Define if PBS Pro support is enabled])
++		  PAC_POP_FLAG(LIBS)
++                  if test "$have_pbs_launcher" = "yes" ; then
++		      available_launchers="$available_launchers pbs"
++		      PAC_APPEND_FLAG([-lpbs],[WRAPPER_LIBS])
++                      AC_DEFINE(HAVE_PBS_PRO, 1, [Define if PBS Pro support is enabled])
++                  fi
+                 fi
+ 		available_rmks="$available_rmks pbs"
+ 		;;

--- a/var/spack/repos/builtin/packages/mvapich2/package.py
+++ b/var/spack/repos/builtin/packages/mvapich2/package.py
@@ -117,6 +117,7 @@ class Mvapich2(AutotoolsPackage):
         values=auto_or_any_combination_of("lustre", "gpfs", "nfs", "ufs"),
     )
 
+    depends_on("automake@1.15", type="build")  # needed for torque patch
     depends_on("findutils", type="build")
     depends_on("bison", type="build")
     depends_on("pkgconfig", type="build")
@@ -134,6 +135,8 @@ class Mvapich2(AutotoolsPackage):
     depends_on("slurm", when="process_managers=slurm")
 
     conflicts("fabrics=psm2", when="@:2.1")  # psm2 support was added at version 2.2
+
+    patch("fix-torque.patch", when="@2.3.7-1")
 
     filter_compiler_wrappers("mpicc", "mpicxx", "mpif77", "mpif90", "mpifort", relative_root="bin")
 

